analyzers/src/SonarAnalyzer.CSharp/Extensions/MethodDeclarationSyntaxExtensions.cs

```diff
@@ -17,13 +17,45 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using StyleCop.Analyzers.Lightup;
 
 namespace SonarAnalyzer.Extensions
 {
     internal static class MethodDeclarationSyntaxExtensions
     {
+        /// <summary>
+        /// Returns true if the method throws exceptions or returns null.
+        /// </summary>
+        public static bool ThrowsOrReturnsNull(this MethodDeclarationSyntax syntaxNode) =>
+            syntaxNode.DescendantNodes().OfType<ThrowStatementSyntax>().Any() ||
+            syntaxNode.DescendantNodes().OfType<ExpressionSyntax>().Any(expression => expression.IsKind(SyntaxKindEx.ThrowExpression)) ||
+            syntaxNode.DescendantNodes().OfType<ReturnStatementSyntax>().Any(returnStatement => returnStatement.Expression.IsKind(SyntaxKind.NullLiteralExpression)) ||
+            // For simplicity this returns true for any method witch contains a NullLiteralExpression but this could be a source of FNs
+            syntaxNode.DescendantNodes().OfType<ExpressionSyntax>().Any(expression => expression.IsKind(SyntaxKind.NullLiteralExpression));
+
+        public static bool IsExtensionMethod(this BaseMethodDeclarationSyntax methodDeclaration) =>
```

Comment:
is there an overlap with https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.CSharp/Extensions/MethodDeclarationExtensions.cs ?

Reply:
@andrei-epure-sonarsource, In this case, @mary-georgiou-sonarsource moved the code from that file to this one. The previous name did not follow our naming convention for extension classes.

Reply:
ah sorry